src/redux/reducer.js

```diff
@@ -683,8 +683,14 @@ export const reducer = createReducer(initialState, {
                             diagram.state = ViewState.MINIMIZED;
                         }
                     });
+                    const diagramToOpen = diagramStates[diagramToOpenIndex];
+
+                    // we push the diagram to the last position, so when we reopen it, it opens at the last position available.
+                    diagramStates.splice(diagramToOpenIndex, 1);
+                    diagramStates.push(diagramToOpen);
+
                     // And update the one to open.
-                    diagramStates[diagramToOpenIndex].state = ViewState.OPENED;
+                    diagramToOpen.state = ViewState.OPENED;
```

Comment:
This line should be moved (see comment above) and the comment should be removed.